core/arch/arm/kernel/thread.c

```diff
@@ -1056,10 +1082,15 @@ void thread_get_user_kdata(struct mobj **mobj, size_t *offset,
 
 	core_mmu_get_user_va_range(&v, NULL);
 	*va = v + thread_user_kcode_size;
+#ifdef CFG_DYN_CONFIG
+	*mobj = thread_user_kdata_page_mobj;
+	*offset = 0;
+#else
 	*mobj = mobj_tee_ram_rw;
-	*sz = sizeof(thread_user_kdata_page);
 	*offset = (vaddr_t)thread_user_kdata_page -
 		  (vaddr_t)mobj_get_va(*mobj, 0, *sz);
+#endif
+	*sz = thread_user_kdata_page_size;
```

Comment:
size if always `ROUNDUP(sizeof(struct thread_core_local) * CFG_TEE_CORE_NB_CORE, SMALL_PAGE_SIZE)`. Is it worth using a memory cell to store it?

Reply:
In a not too distant future it will become `ROUNDUP(sizeof(struct thread_core_local) * thread_core_count, SMALL_PAGE_SIZE)`. I can wrap it in a helper function instead if you prefer that.